### PR TITLE
Widget classes: Checkout widgets

### PR DESCRIPTION
**Pull request by felixgehrmann**
### All changes meet the following requirements
- [ ] Changelog entry was added
- [ ] Changes have been tested by the author
- [ ] Changes have been tested by the reviewer
- [ ] Plugin can be built

@plentymarkets/ceres-io

### CODE REVIEW

**Comment by felixgehrmann** [comment]
Work on widget group is mostly done, requires features in SettingFactories to be completed.
#1661 for maxPerPage WidgetData
Ability to add tooltips is missing from SettingFactories.

**Comment by felixdausch** [comment]
With the changes of https://github.com/plentymarkets/plugin-ceres/pull/1659 you can use `ValueListFactory::make()` for more streamlined code

**Comment by felixgehrmann** [comment]
Ready for review.

File: src/Widgets/Checkout/CancelPaymentWidget.php [hunk 1]
```diff
@@ -3,8 +3,50 @@
...
 {
     protected $template = "Ceres::Widgets.Checkout.CancelPaymentWidget";
+
+    public function getData()
+    {
+        return WidgetDataFactory::make("Ceres::CancelPaymentWidget")
+            ->withLabel("Widget.cancelPaymentLabel")
+            ->withPreviewImageUrl("/images/widgets/cancel-payment.svg")
+            ->withType(WidgetTypes::CHECKOUT)
+            ->withCategory(WidgetCategories::BASKET)
```
Comment (felixdausch) [inline src/Widgets/Checkout/CancelPaymentWidget.php]:
```suggestion
            ->withCategory(WidgetCategories::CHECKOUT)
```

File: src/Widgets/Checkout/CancelPaymentWidget.php [hunk 2]
```diff
@@ -3,8 +3,50 @@
...
+
+    public function getSettings()
+    {
+        /** @var WidgetSettingsFactory $settings */
+        $settings = pluginApp(WidgetSettingsFactory::class);
+
+        $settings->createCustomClass();
+        $settings->createAppearance();
+
+        $select = $settings->createSelect("buttonSize")
```
Comment (felixdausch) [inline src/Widgets/Checkout/CancelPaymentWidget.php]: see https://github.com/plentymarkets/plugin-ceres/pull/1676

File: src/Widgets/Checkout/CancelPaymentWidget.php [hunk 3]
```diff
@@ -3,8 +3,50 @@
...
+            ->withDefaultValue("")
+            ->withName("Widget.widgetButtonSizeLabel")
+            ->withTooltip("Widget.widgetButtonSizeTooltip");
+
+        $select->withListBoxValues(
```
Comment (felixdausch) [inline src/Widgets/Checkout/CancelPaymentWidget.php]:
```suggestion
        ->withListBoxValues(
```

File: src/Widgets/Checkout/GtcCheckWidget.php [hunk 4]
```diff
@@ -3,8 +3,58 @@
...
     protected $template = "Ceres::Widgets.Checkout.GtcCheckWidget";
+
+    public function getData()
+    {
+        return WidgetDataFactory::make("Ceres::GtcCheckWidget")
+            ->withLabel("Widget.gtcCheckLabel")
+            ->withPreviewImageUrl("/images/widgets/gtc-check.svg")
+            ->withType(WidgetTypes::CHECKOUT)
+            ->withCategory(WidgetCategories::CHECKOUT)
+            ->withPosition(400)
```
Comment (felixdausch) [inline src/Widgets/Checkout/GtcCheckWidget.php:22]: `maxPerPage: 1` fehlt!

File: src/Widgets/Checkout/PlaceOrderWidget.php [hunk 5]
```diff
@@ -3,8 +3,51 @@
...
+    public function getSettings()
+    {
+        /** @var WidgetSettingsFactory $settings */
+        $settings = pluginApp(WidgetSettingsFactory::class);
+
+        $settings->createCustomClass();
+        $settings->createAppearance()
+            ->withDefaultValue("success");
+
+        $select = $settings->createSelect("buttonSize")
```
Comment (felixdausch) [inline src/Widgets/Checkout/PlaceOrderWidget.php]: see https://github.com/plentymarkets/plugin-ceres/pull/1676

**Review by felixdausch — CHANGES_REQUESTED** [review]

**Review by felixdausch — APPROVED** [review]

**Pull request merged**